src/interpreter.c

```diff
@@ -756,20 +758,31 @@ JL_DLLEXPORT size_t jl_capture_interp_frame(jl_bt_element_t *bt_entry,
         void *stateend, size_t space_remaining)
 {
     interpreter_state *s = &((interpreter_state*)stateend)[-1];
-    int need_module = !s->mi;
-    int required_space = need_module ? 4 : 3;
+    int need_module = !s->mi_or_method;
+    int is_opaque_closure = s->mi_or_method && jl_is_method(s->mi_or_method);
+    size_t njlvalues = need_module ? 2 : 1;
+    // For opaque closure, we also capture the arguments in order to be able to
+    // say something about them (in compiled code, the compiler would have
+    // synthesized a MethodInstance with an appropriate specialization).
+    if (is_opaque_closure) {
+        njlvalues += ((jl_method_t*)s->mi_or_method)->nargs;
```

Comment:
I thought `njlvalues` in this function has a maximum value, so cannot be populated with an undeterminant list. Perhaps we should allocate an args_type_tuple here instead, and store that to the backtrace frame?

Reply:
I was thinking about that, but I believe this entire function is JL_NOT_SAFEPOINT

Reply:
Hm, that is awkward to support then, since it is a fixed maximum frame size. Unless the caller makes the svec for us.